### PR TITLE
Switch e2e test `docker-compose` to `docker compose`

### PR DESCRIPTION
**Pull request by steiza**
#### Summary

Many cosign pull requests have e2e test failing, such as https://github.com/sigstore/cosign/actions/runs/10238677353/job/28323361888?pr=3808.

Apparently this is because `docker-compose` was removed from the ubuntu-latest and windows GitHub Actions runners late last week:

https://github.com/actions/runner-images/issues/9692

I don't follow the ecosystem closely enough to understand it, but apparently `docker compose` is what most people are using to work around this.

#### Release Note

NONE

#### Documentation

N/A

### CODE REVIEW

**Comment by steiza** [comment]
Let's work on https://github.com/sigstore/cosign/pull/3809 instead.

File: test/e2e_test.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ set -ex
 echo "setting up OIDC provider"
 pushd ./test/fakeoidc
 oidcimg=$(ko build main.go --local)
-docker network ls | grep fulcio_default || docker network create fulcio_default
+docker network ls | grep fulcio_default || docker network create --label com.docker.compose.network=fulcio_default fulcio_default
```
Comment (cmurphy) [inline test/e2e_test.sh:22]: This works for me if I have

```suggestion
docker network ls | grep fulcio_default || docker network create --label com.docker.compose.network=default fulcio_default
```

**Review by MMONSTER124 — CHANGES_REQUESTED** [review]
Solicitação de pull

**Review by MMONSTER124 — COMMENTED** [review]
ba500718fe3ab16dc64a845542e2d6e229d7e5def3d823960791ff32b9d56597
main.go

**Pull request closed without merging**